lib/archethic/p2p/message/dashboard_data.ex

```diff
@@ -27,20 +27,20 @@ defmodule Archethic.P2P.Message.DashboardData do
           <<DateTime.to_unix(datetime)::32, count::bitstring, durations_serialized::bitstring>>
       end)
 
-    buckets_count = VarInt.from_value(map_size(buckets))
+    buckets_count = VarInt.from_value(length(buckets))
 
     <<buckets_count::bitstring, buckets_serialized::bitstring>>
   end
 
   @spec deserialize(bitstring()) :: {t(), bitstring}
   def deserialize(<<rest::bitstring>>) do
     {buckets_count, rest} = VarInt.get_value(rest)
-    {buckets, rest} = deserialize_buckets(buckets_count, %{}, rest)
+    {buckets, rest} = deserialize_buckets(buckets_count, [], rest)
 
     {%__MODULE__{buckets: buckets}, rest}
   end
 
-  defp deserialize_buckets(0, acc, rest), do: {acc, rest}
+  defp deserialize_buckets(0, acc, rest), do: {Enum.reverse(acc), rest}
```

Comment:
Suggestion: To optimize the computation, I think you can omit `Enum.reverse/1` as the order is not required outside the serialization